batchflow/research/workers.py

```diff
@@ -174,7 +174,13 @@ def init(self):
         """ Run before job execution. """
         i, job = self.job
         n_branches = len(job.configs)
-        self.device_configs = [self.devices[i] for i in range(n_branches)]
+        mapping = {
+            item: i
+            for i, item in enumerate({self.devices[i]['device'] for i in range(n_branches)})
+        }
+        os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
+        os.environ["CUDA_VISIBLE_DEVICES"] = ','.join([self.devices[i]['device'] for i in range(n_branches)])
```

Comment:
in n_branches > number of real devices, there will be repeating values, right? Is it OK?

Reply:
Yes